pallets/xcm-core-buyer/src/lib.rs

```diff
@@ -640,44 +700,31 @@ pub mod pallet {
         type Call = Call<T>;
 
         fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
-            if let Call::buy_core { para_id, proof } = call {
-                /*
-                if <Pallet<T>>::is_online(heartbeat.authority_index) {
-                    // we already received a heartbeat for this authority
-                    return InvalidTransaction::Stale.into()
+            if let Call::buy_core {
+                para_id,
+                collator_account_id,
+                proof,
+            } = call
+            {
+                let block_number = <frame_system::Pallet<T>>::block_number();
```

Comment:
I added the block number to be able to do

```
.and_provides((block_number, para_id))
```

because without that, only a single unsigned transaction would ever be included by the block builder. Now that we have the nonce we probably don't need it. But I don't know a good alternative, `(para_id, nonce)` won't work because the nonce can be the same for different collators... So probably it's better to just leave it like it is. And unfortunately this is going to be very hard to test...